src/common.cpp

```diff
@@ -1873,6 +1884,63 @@ std::string get_executable_path(const char *argv0) {
     } else {
         return std::string(buff);
     }
+#elif defined(__OpenBSD__)
+    static std::string result;
+    if (!result.empty()) return result;
+    int cntp = 0;
+    struct stat st;
+    bool ok = false;
+    std::string path;
+    kinfo_file *kif = nullptr;
+    static kvm_t *kd = nullptr;
+    char errbuf[_POSIX2_LINE_MAX];
+    const char *pwd = nullptr, *penv = nullptr;
```

Comment:
You only need these in a small block, you can define them there:

```c++
            const char *penv = getenv("PATH");
```

Reply:
Same goes for the other variables - no need to define `kinfo_file *kif` twenty lines before you need it. This isn't C89, you can define variables later.